### DISCUSSION TITLE
b-dropdown emit duplicate events when changing page

### DISCUSSION THREAD

**Discussion by hockeylagu** [bootstrap-vue/bootstrap-vue#5811] [Q&A] [answered]
### Describe the bug

On the initial load of the page the dropdown emit only once and that's the expected bethavior. But when we change the page with vue-router the dropdown will emit the "show" multiple times.

### Steps to reproduce the bug

1. Go here: https://o024d.csb.app/ or here: https://codesandbox.io/s/young-forest-o024d
2. Open the devtool
3. Click on the dropdown
4. Click on the about link
5. Click on the home link
6. Click on the dropdown
7. Multiple emit in the console for the show event

### Expected behavior

The emit should only be emitted once.

### Versions

**Libraries:**

- BootstrapVue: 2.17.3
- Bootstrap: 4.#.#
- Vue: 2.6.11

**Environment:**

- OS: Windows 10
- Browser: Chrome

### Demo link

[CodeSandbox](https://codesandbox.io/s/young-forest-o024d?file=/src/App.vue)

**Comment by Dan-Correia** [comment, marked as answer]
The reason why this is happening is because you're adding an event handler every time the component with the dropdown is rendered. 

So you either remove the event handler on the `beforeDestroy()` lifecycle event or you just call your method on the `show()` event on the actual bootstrap vue's dropdown element.

Here's a working version using the code you've written - [https://codesandbox.io/s/mystifying-haslett-jzc07](https://codesandbox.io/s/mystifying-haslett-jzc07?file=/src/views/Home.vue )
